app/src/main/java/com/example/squareonetestproject/MainActivity.kt

```diff
@@ -22,25 +58,98 @@ class MainActivity : ComponentActivity() {
+        items(movies) { movie ->
+
+            var showDetails by remember { mutableStateOf(false) }
+
+            if(showDetails) Dialog(onDismissRequest = { showDetails = false }) {
+                Card {
+                    LazyColumn {
+                        item {
+                            Column {
+                                AsyncImage(
+                                    model = movie.poster.create(LocalContext.current),
+                                    contentDescription = null,
+                                    modifier = Modifier.fillMaxWidth()
+                                )
+
+                                Text(text = movie.name)
+                                Text(text = movie.overview)
+                                Text(text = movie.releaseDate)
+                                Text(text = movie.genresList.contentToString())
+                            }
+                        }
+
+                    }
+
+                }
+                movie.name
+            }
+            Card(
+                modifier = Modifier.padding(10.dp),
+                onClick = { showDetails = true} ) {
+                AsyncImage(
+                    model = movie.poster.create(LocalContext.current),
+                    contentDescription = null,
+                    modifier = Modifier.fillMaxWidth()
+                )
+                Text(
+                    modifier = Modifier.align(Alignment.CenterHorizontally),
+                    text = movie.name)
+            }
+
+        }
+    }
 }
 
-@Preview(showBackground = true)
-@Composable
-fun GreetingPreview() {
-    SquareOneTestProjectTheme {
-        Greeting("Android")
+/**
+ *
+ */
+class MovieDetail(
+    val name: String,
+    val poster: ImageRequestFactory,
+    val genresList: Array<Int>,
+    val overview: String,
+    val releaseDate: String
+) {
+    /**
+     * It could be useless to create a factory for this feature, but
+     * at the very first I thought I have to carry the authorization header so I
+     * made this class to provide advanced features. But it does not have any security checks...
+     */
+    class ImageRequestFactory(
+        private val url: String,
+        private val authToken: String
+    ) {
+        fun create(context: Context): ImageRequest {
+            return ImageRequest.Builder(context)
+                .addHeader("accept", "application/json")
+                .addHeader("Authorization", "Bearer $authToken")
+                .data(url)
+                .build()
+
+        }
     }
+}
+
+class MainViewModel(useCase: UseCase) : ViewModel() {
+    class Factory(private val useCase: UseCase): ViewModelProvider.Factory {
+        override fun <T : ViewModel> create(modelClass: Class<T>): T {
+            return MainViewModel(useCase) as T
+        }
+    }
+
+    val movies = useCase.getMovies().stateIn(
+        viewModelScope, SharingStarted.Lazily, listOf())
```

Comment:
Any special reason to do it manually instead of using a dependency injection tool?

Reply:
No special reason. Currently it is using Hilt